### ISSUE TITLE
Use views container to add Azure section in activity bar

### ISSUE THREAD

**Issue by fiveisprime** [Microsoft/vscode-azuretools#167]
See [vscode#43645](https://github.com/Microsoft/vscode/issues/43645)

Contribute to a new Azure section in the activity bar rather than individual explorers in the files view. The functionality in VS Code will ship in the next release (sometime this week) so let's start thinking through this.

I'm not incredibly concerned about the tremendous shift in location given there will be a new Azure icon in the sidebar and we may be one of the first extensions to contribute to the activity bar. I expect this could be an issue once the bar starts to get cluttered.

**Comment by EricJizbaMSFT** [comment]
@fiveisprime I don't think there's a way to support old _and_ new versions of VS Code with this. As far as I can tell, we have to replace 'explorer' with 'azure' in the following code (which would break people on older version of VS Code):
```json
"views": {
    "explorer": [
        {
            "id": "azureAppService",
            "name": "Azure App Service",
            "when": "config.appService.showExplorer == true"
        }
    ]
}
```

Unless there's some magic logic we can do in the 'when' clause that I'm not aware of?

**Comment by fiveisprime** [comment]
Dang! Any ideas @chrisdias?

**Comment by chrisdias** [comment]
CC @sandy081 

You can specify the min version of VS Code required by setting the engine to `^1.23.0` such that the extension will only be installed/updated for users that have that version (or greater):

In `package.json`:

``` json
  "engines": {
    "vscode": "^1.23.0"
  }
```

**Comment by EricJizbaMSFT** [comment]
Yes @chrisdias I'm aware of that, but we need to ship App Service before //build and I'm trying to plan for the (hopefully unlikely) case that VS Code doesn't push this update in time

**Comment by chrisdias** [comment]
@EricJizbaMSFT we're hoping to ship VS Code 1.23 this week.

**Comment by EricJizbaMSFT** [comment]
Okey dokey! I suppose it's not too hard to ship the vsix from right before this change if VS Code slips anyways

**Comment by EricJizbaMSFT** [comment]
@StephenWeatherford @fiveisprime @chrisdias Do you think the JSON Outline for the ARM extension should also be moved to the 'Azure' view? I'm leaning no - since it's directly related to the _file_ you have open, which to me feels appropriate for the file explorer container (Also it doesn't show any remote Azure resources)

**Comment by StephenWeatherford** [comment]
Good point, I'm inclined to agree, at least until we've played with it a bit.

**Comment by fiveisprime** [comment]
There will be a note about view containers in the VS Code release notes and people read those. That, along with the fact there will be a new icon in the activity view makes me think people will figure it out pretty quickly.
